tests/baselines/reference/jsFileMethodOverloads.types

```diff
@@ -0,0 +1,76 @@
+=== tests/cases/compiler/jsFileMethodOverloads.js ===
+/**
+ * @template T
+ */
+ class Example {
+>Example : Example<T>
+
+  /**
+   * @param {T} value 
+   */
+  constructor(value) {
+>value : T
+
+    this.value = value;
+>this.value = value : T
+>this.value : any
+>this : this
+>value : any
+>value : T
+  }
+
+  /**
+   * @overload
+   * @this {Example<'number'>}

```

Comment:
I don't see `this` in the type. Does this actually work?

Reply:
Thank you for pointing this out! I forgot that `JSDocSignature` does not support `@this` tag, but it seems like `@param {...} this` works well. I've just pushed a fix.